baselines/flanders/flanders/strategy.py

```diff
@@ -0,0 +1,419 @@
+"""FLANDERS strategy."""
+
+import typing
+from logging import INFO, WARNING
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+from flwr.common import (
+    EvaluateIns,
+    EvaluateRes,
+    FitIns,
+    FitRes,
+    MetricsAggregationFn,
+    NDArrays,
+    Parameters,
+    Scalar,
+    ndarrays_to_parameters,
+    parameters_to_ndarrays,
+)
+from flwr.common.logger import log
+from flwr.server.client_manager import ClientManager
+from flwr.server.client_proxy import ClientProxy
+from flwr.server.strategy.aggregate import aggregate, weighted_loss_avg
+
+from .utils import load_all_time_series
+
+WARNING_MIN_AVAILABLE_CLIENTS_TOO_LOW = """
+Setting `min_available_clients` lower than `min_fit_clients` or
+`min_evaluate_clients` can cause the server to fail when there are too few clients
+connected to the server. `min_available_clients` must be set to a value larger
+than or equal to the values of `min_fit_clients` and `min_evaluate_clients`.
+"""
+
+
+class Flanders:
```

Comment:
Wouldn't it be better to make `Flanders` be a strategy class inheriting from Flower's `FedAvg`? This would make your baseline easier to maintain. Also, you wouldn't need to re-implement `num_fit_clients`, `num_evaluation_clients`, `initialize_parameters`, `evaluate` and `aggregate_evaluate` evaluate methods...

Reply:
That what I used to do, but mypy gets angry because some methods do not have the same signature of FedAvg (if I recall correctly, the error was ` error: Signature of [method] incompatible with supertype "FedAvg".`). I think that it thinks that I am overriding, but I'm actually overloading and it should be a safe operation AFAIK

Reply:
ok. from a quick look at this, it seems the only source of incompatibility is passing `client_state` to `aggregate_fit`. You could pass this from your server using a new method for example (that can set a `self.client_state` inside the strategy and then access that when needed in `aggregate_fit`)

Reply:
I realised that `client_state` in `aggregate_fit` was a typo. Now only the `evaluate` method is an overloading. I managed to solve the issue with mypy by using keyword arguments with default values in the signature, but pylint still complains: `W0221: Parameters differ from overridden 'evaluate' method (arguments-differ)`. At this point either I ignore the warning or I implement a new way of saving the results (the way the evaluate methods are designed are to first evaluate the model and then save the results in the csv file, that's why I need two more arguments). Maybe I can get this working by using the history returning from `start simulation()`... but I'm not sure.